### DISCUSSION TITLE
Localization: Warning: Prop `href` did not match.

### DISCUSSION THREAD

**Discussion by cibulka** [vercel/next.js#36104] [Help] [answered]
Hi everyone! 

My language switcher component is giving me the SSR mismatch error: `Warning: Prop 'href' did not match. Server: "" Client: "/about-us"`. 

It is however quite weird, as the dev process (console) is giving me same information in console.log:

```
console.log(router.route, router.query, router.locale)
=> browser: about-us {} en
=> server: about-us {} en
```

The components goes like this:

```
import React, { FC } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';

const LocaleLink: FC<{ className?: string }> = (props) => {
  const { locale, route, query } = useRouter();

  let otherLocale;
  if (locale === 'cs') otherLocale = 'en';
  if (locale === 'en') otherLocale = 'cs';

  console.log(route, query, locale);

  return otherLocale ? (
    <Link href={{ href: route, query }} locale={otherLocale}>
      <a className={props.className}>
        {locale === 'cs' && 'Switch to English'}
        {locale === 'en' && 'Přepnout do češtiny'}
      </a>
    </Link>
  ) : null;
};

export default LocaleLink;
```

What am I doing wrong here? 

Maybe there is some other way recommended way how to build a "localed" links? They need to include the full query though (such as `/cs/product/10?color=red` and `/en/product/10?color=red`). 

Thank you!

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Why are you doing:

```jsx
<Link href={{ href: route, query }} locale={otherLocale}>
```

`href` is not part of the `href` prop, `pathname` is:

```jsx
  <Link
    href={{
      pathname: route,
      query,
    }}
    locale={otherLocale}
  >
```

I'd also recommend using `pathname`, or `asPath` from `useRouter`:


```jsx
 <Link href={asPath} locale={otherLocale}>
```

or

```jsx
  <Link
    href={{
      pathname,
      query,
    }}
    locale={otherLocale}
  >
```

Either way, using `href={{pathname: '/something', query}}` will fix the error you are seeing. See more in the [docs](https://nextjs.org/docs/api-reference/next/link#with-url-object).

**Reply by cibulka** [reply]
@icyJoseph [icyJoseph](https://github.com/icyJoseph) - sorry for the late reply, also it seems you were right :) thank you!

Would be nice if Next.js and/or TS gave me a clearer error message however. :/
